bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieNettyServer.java

```diff
@@ -84,24 +105,42 @@
         requestDecoder = new BookieProtoEncoding.RequestDecoder(registry);
 
         if (!conf.isDisableServerSocketBind()) {
-            channels.add(new NioServerSocketChannelManager());
+            ThreadFactory threadFactory = new ThreadFactoryBuilder().setNameFormat("bookie-io-%s").build();
+            final int numThreads = Runtime.getRuntime().availableProcessors() * 2;
+
+            EventLoopGroup eventLoopGroup;
+            if (SystemUtils.IS_OS_LINUX) {
+                try {
+                    eventLoopGroup = new EpollEventLoopGroup(numThreads, threadFactory);
+                } catch (Throwable t) {
```

Comment:
In Pulsar we are catching multiple exceptions: 

`ExceptionInInitializerError | NoClassDefFoundError | UnsatisfiedLinkError e`